homeassistant/components/media_player/__init__.py

```diff
@@ -915,7 +938,16 @@ async def get(self, request: web.Request, entity_id: str) -> web.Response:
         if not authenticated:
             return web.Response(status=HTTP_UNAUTHORIZED)
 
-        data, content_type = await player.async_get_media_image()
+        media_content_type = request.query.get("media_content_type")
+        media_content_id = request.query.get("media_content_id")
+        browse_image = request.query.get("browse_image")
```

Comment:
Why do we need to pass `browse_image` if we already pass media content `id` and `type`?

Reply:
there are cases where there is a second identifier like thex8

Reply:
we could make this browse_image optional as i think most integrations could determine the image using media_content_id

Reply:
Yes, it should be optional. And we should name it better. For example, we should make sure it never ever contains a url, as that allows for injection attacks where you can trick Home Assistant into fetching a url (with the privileges that come from the HA host) and pass it on to the user.

Reply:
I think that we should stop overloading this webview to do two things, that's weird. We should register an extra url. This can be done with `extra_urls` prop on HTTPView.